scripts/Muon/GUI/Common/fitting_tab_widget/fitting_tab_presenter.py

```diff
@@ -44,7 +61,82 @@ def handle_use_rebin_changed(self):
         if not self.manual_selection_made:
             self.update_selected_workspace_guess()
         else:
-            self.selected_data = self.context.get_list_of_binned_or_unbinned_workspaces_from_equivalents(self.selected_data)
+            self.selected_data = self.context.get_list_of_binned_or_unbinned_workspaces_from_equivalents(
+                self.selected_data)
+
+    def handle_fit_type_changed(self):
+        fit_type = self.view.fit_type
+
+        if fit_type == 'Single Fit':
+            self.view.workspace_combo_box_label.setText('Select Workspace')
+            if not self.manual_selection_made:
+                guess_selection = self.context.get_names_of_workspaces_to_fit(runs='All',
+                                                                              group_and_pair='All', phasequad=True,
+                                                                              rebin=not self.view.fit_to_raw)
+            else:
+                guess_selection = self.selected_data
+            self.selected_data = guess_selection
+        else:
+            self.view.workspace_combo_box_label.setText('Display parameters for')
+            if not self.manual_selection_made:
+                guess_selection = self.context.get_names_of_workspaces_to_fit(runs='All',
+                                                                              group_and_pair=self.context.gui_context[
+                                                                                  'selected_group_pair'], phasequad=True,
+                                                                              rebin=not self.view.fit_to_raw)
+            else:
+                guess_selection = self.selected_data
+
+            self.selected_data = guess_selection
+
+    def handle_fit_clicked(self):
+        fit_type = self.view.fit_type
+
+        try:
+            if fit_type == 'Single Fit':
+                single_fit_parameters = self.get_parameters_for_single_fit()
+                self.model.do_single_fit(single_fit_parameters)
+            elif fit_type == 'Simultaneous Fit':
+                simultaneous_fit_parameters = self.get_multi_domain_fit_parameters()
+                self.model.do_simultaneous_fit(simultaneous_fit_parameters)
+            elif fit_type == 'Sequential Fit':
+                sequential_fit_parameters = self.get_multi_domain_fit_parameters()
+                self.model.do_sequential_fit(sequential_fit_parameters)
+        except ValueError as e:
+            self.view.warning_popup(e)
+
+    def handle_start_x_updated(self):
+        value = self.view.start_time
+        index = self.view.get_index_for_start_end_times()
+        self.update_start_x(index, value)
+
+    def handle_end_x_updated(self):
+        value = self.view.end_time
+        index = self.view.get_index_for_start_end_times()
+        self.update_end_x(index, value)
+
+    def get_parameters_for_single_fit(self):
+        params = {}
```

Comment:
lets call a common function to set the parameters that are the same e.g. Function